compiler/rustc_codegen_llvm/src/llvm/mod.rs

```diff
@@ -18,13 +17,13 @@ use crate::common::AsCCharPtr;
 
 pub(crate) mod archive_ro;
 pub(crate) mod diagnostic;
-pub(crate) mod enzyme_ffi;
-mod ffi;
 
-pub(crate) use self::enzyme_ffi::*;
+pub(crate) trait LLVMRustResultExt {
+    fn into_result(self) -> Result<(), ()>;
+}
 
-impl LLVMRustResult {
-    pub(crate) fn into_result(self) -> Result<(), ()> {
+impl LLVMRustResultExt for LLVMRustResult {
+    fn into_result(self) -> Result<(), ()> {
```

Comment:
I think this could move to `rustc_llvm`, maybe in `mod impls`, so it's still an inherent method. Then you won't need the `use ...Ext as _` additions.